cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/MessageBatchQueryProvider.java

```diff
@@ -157,9 +157,9 @@ private Select addFilter(Select select, StoredMessageFilter filter)
 		if (!isLeftIndexSelected && filter.getTimestampFrom() != null)
 			select = FilterUtils.filterToWhere(ComparisonOperation.GREATER_OR_EQUALS, select.whereColumn(MESSAGE_INDEX),
 					LEFT_MESSAGE_INDEX);
-		
+
 		if (!isRightIndexSelected && filter.getTimestampTo() != null)
-			select = FilterUtils.filterToWhere(ComparisonOperation.LESS_OR_EQUALS, select.whereColumn(MESSAGE_INDEX),
+			select = FilterUtils.filterToWhere(filter.getTimestampTo().getOperation(), select.whereColumn(MESSAGE_INDEX),
```

Comment:
With `filter.getTimestampTo().getOperation()` you can miss a batch if `timestamp` is the middle of the batch.